src/provisioning-ofono.c

```diff
@@ -392,8 +426,19 @@ static void add_internet_context_reply(DBusPendingCall *call, void *user_data)
 	DBusMessage *reply = dbus_pending_call_steal_reply(call);
 	DBusError err;
 	const char *path;
-	struct modem_data *modem;
+	struct modem_data *modem = NULL;
 	struct provisioning_data *prov_data;
+	GList *keys = g_hash_table_get_keys(modem_list);
+
+	while (keys != NULL) {
+		modem = g_hash_table_lookup(modem_list, keys->data);
+
+		if (modem->gprs_attached) {
+			break;
+		}
+		
+		keys = keys->next;
+	}
 
```

Comment:
This code was added to fetch the modem having valid data connection as earlier code was referencing a global variable and wrong modem was updated with provisioning settings